cmd/snap-confine/snap-confine.c

```diff
@@ -681,9 +687,11 @@ static void enter_non_classic_execution_environment(sc_invocation * inv,
 	}
 	if (!sc_cgroup_is_v2()) {
 		sc_cgroup_freezer_join(inv->snap_instance, getpid());
-		if (sc_feature_enabled(SC_FEATURE_REFRESH_APP_AWARENESS)) {
-			sc_cgroup_pids_join(inv->security_tag, getpid());
-		}
+	}
+	// TODO: This should be done uniformly for classic and non-classic snaps.
+	// It will be untangled in the next pass.
+	if (sc_feature_enabled(SC_FEATURE_REFRESH_APP_AWARENESS)) {
```

Comment:
Looking at it, I think we should join the hierarchy right now. Otherwise, even when this lands, we won't be able to benefit on a system with unified hierarchy, and we already know of bugs like this one https://forum.snapcraft.io/t/cant-open-any-other-apps-like-browsewr-when-clicking-link-if-installed-on-snap-fedora/13524

Reply:
We discussed this and I agree conceptually but disagree about joining it unconditionally via this PR. I think it warrants a separate discussion.

Reply:
I tend to agree with @zyga on this, I think unconditionally joining it should be a separate discussion

Reply:
Ok, let's get this landed first and then we'll come back to the discussion.